spyder/api/widgets/main_widget.py

```diff
@@ -903,6 +924,15 @@ def on_close(self):
         """
         pass
 
+    def on_focus_in(self):
+        """Perform actions when the widget is focused."""
+        pass
+
+    def on_focus_out(self):
+        """Perform actions when the widget is no longer focused."""
+        pass
```

Comment:
Since this are new methods probably we should bump the API version number (`spyder/api/_version.py`) to be `0.8.0` (maybe we should also add some sort of comment there regarding the rules to modify that number)

Reply:
See https://github.com/spyder-ide/spyder/pull/17467/commits/8fd758e46f3450262f623ee0d3a00efe54b62b36